test/UndisclosedPasswordTest.php

```diff
@@ -0,0 +1,233 @@
+use Psr\Http\Message\ResponseInterface;
+use Zend\Validator\UndisclosedPassword;
+use ZendTest\Validator\TestAsset\HttpClientException;
+
+class UndisclosedPasswordTest extends TestCase
+{
+    /**
+     * @var ClientInterface
+     */
+    private $httpClient;
+
+    /**
+     * @var RequestInterface
+     */
+    private $httpRequest;
+
+    /**
+     * @var ResponseInterface
+     */
+    private $httpResponse;
+
+    /**
+     * @var UndisclosedPassword
+     */
+    private $validator;
+
+    /**
+     * @inheritDoc
+     */
+    public function setUp()
+    {
+        $this->httpClient = $this->getMockBuilder(ClientInterface::class)
+            ->getMockForAbstractClass();
+        $this->httpRequest = $this->getMockBuilder(RequestFactoryInterface::class)
+            ->getMockForAbstractClass();
+        $this->httpResponse = $this->getMockBuilder(ResponseInterface::class)
+            ->getMockForAbstractClass();
+        $responseFactoryInterface = $this->getMockBuilder(ResponseFactoryInterface::class)
+            ->getMockForAbstractClass();
+
+        $this->validator = new UndisclosedPassword(
+            $this->httpClient,
+            $this->httpRequest,
+            $responseFactoryInterface
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    protected function tearDown()
+    {
+        $this->httpClient = null;
+    }
+
+    /**
+     * Data provider returning good, strong and unseen
+     * passwords to be used in the validator.
+     *
+     * @return array
+     */
+    public function goodPasswordProvider()
+    {
+        return [
+            ['ABi$B47es.Pfg3n9PjPi'],
+            ['potence tipple would frisk shoofly'],
+        ];
+    }
+
+    /**
+     * Data provider for most common used passwords
+     *
+     * @return array
+     * @see https://en.wikipedia.org/wiki/List_of_the_most_common_passwords
+     */
+    public function seenPasswordProvider()
+    {
+        return [
+            ['123456'],
+            ['password'],
+            ['123456789'],
+            ['12345678'],
+            ['12345'],
+        ];
+    }
+
+    /**
+     * Testing that we reject invalid password types
+     *
+     * @covers \Zend\Validator\UndisclosedPassword::__construct
+     * @covers \Zend\Validator\UndisclosedPassword::isValid
+     * @covers \Zend\Validator\AbstractValidator::createMessage
+     * @covers \Zend\Validator\AbstractValidator::error
+     * @todo Can be replaced by a \TypeError being thrown in PHP 7.0 or up
+     */
+    public function testValidationFailsForInvalidInput()
+    {
+        $this->assertFalse($this->validator->isValid(true));
+        $this->assertFalse($this->validator->isValid((new \stdClass())));
+        $this->assertFalse($this->validator->isValid(['foo']));
+    }
+
+    /**
+     * Test that a given password was not found in the HIBP
+     * API service.
+     *
+     * @param string $password
+     *
+     * @covers \Zend\Validator\UndisclosedPassword::__construct
```

Comment:
Consider using a single `@covers \Zend\Validator\UndisclosedPassword` at class level: this sort of fine-grained coverage is more added maintenance work when someone splits a `private` detail into more `private` methods